### ISSUE TITLE
Unexpected labels for alerts in `/api/prometheus/grafana/api/v1/alerts`

### ISSUE THREAD

**Issue by codesome** [grafana/grafana#45883]
**What happened**:

*This is for Prometheus alert-generator compliance, read more [here](https://github.com/prometheus/compliance/blob/main/alert_generator/specification.md). Using Grafana managed alert on a Prometheus datasource.*

A compliant behaviour would only expect the labels coming from the alerting rule query and labels from rule configuration attached to the alerts in `/api/prometheus/grafana/api/v1/alerts`.

Grafana attaches these additional labels which should not be present in that API:
* `__alert_rule_namespace_uid__`
* `__alert_rule_uid__`
* `datasource_uid`
* `ref_id`

**What you expected to happen**:

Those labels not present for the alerts in `/api/prometheus/grafana/api/v1/alerts`.

**Anything else we need to know?**:

This is mainly for how the API outputs the data. The internals of the alerting engine might not require any changes for this.

**Environment**:
- Grafana version: 8.4.2

**Comment by gotjosh** [comment]
Would this be OK if we supported the hide of private labels via a query param? Say, something like: `/api/prometheus/grafana/api/v1/alerts?privateLabels=false`

**Comment by RichiH** [comment]
The default behavior should be compliant. Headers seem cleaner, but GET also OK.

**Comment by RichiH** [comment]
That would be better IMO, yes.

Sent by mobile; please excuse my brevity.

**Comment by gotjosh** [comment]
@gillesdemey I don't think this is a problem for the front-end, correct? If my understand is correct, we hide private labels from alert instances anyways but unsure if we use them for any matching.

If so, we'd need to change the front-end so that they request can come with `privateLabels=true` on the case of Grafana.

**Comment by gillesdemey** [comment]
Correct, we always hide private labels on the front-end and we don't use them for business logic 👍

**Comment by gillesdemey** [comment]
@gotjosh this sort of got lost in my inbox; I'll take a stab at removing it and see if I can yell louder than my IDE.

**Comment by gillesdemey** [comment]
I turns out we can't remove this code (yet) – sometimes the `AlertLabels` list is being fed data from endpoints like `/api/v2/alerts/groups` that do not omit the private labels 😞
